src/linux.rs

```diff
@@ -136,6 +137,22 @@ impl Command {
         self
     }
 
+    /// Reassociate child process with a namespace specified by a file
+    /// descriptor
+    ///
+    /// `fd` argument is a file descriptor referring to a namespace
+    ///
+    /// 'ns' is an optional namespace type. `None` allows any type of
+    /// namespace
+    ///
+    /// See `man 2 setns` for further details
+    pub fn setns(&mut self, fd: RawFd, ns: Option<Namespace>)
+        -> &mut Command
+    {
+        self.config.setns_namespaces.push((fd, ns));
```

Comment:
There are two problems with this line:

1. We want a safe interface. It means we must ensure that file descriptor is alive as long as the command is alive. Which means two things:
 a) we must duplicate FD, or accept `IntoRawFd` object instead (which ensures that we get owned descriptor in result)
 b) we must close file descriptor after command is dropped (in parent process)

2. We usually have setter methods which replace the value instead of appending to the list. I mean if there is no sense in `x.setns(1, Namespace::Network).setns(2, Namespace::Network)`, the second call should make the first one useless. Just like `x.env("A", "B").env("A", "C")` leaves `C` in `A`. Instead of making a messy env with a duplicated environment variables. While I undrestand that `None` as `ns` ruins the whole idea of making namespaces a dictionary, I'd rather drop it to make semantics more clear. Because I don't have any actually useful use cases for `setns(x, None)` (except for lazyness)